daemon/info.go

```diff
@@ -236,8 +242,19 @@ func memInfo() *system.MemInfo {
 	return memInfo
 }
 
-func operatingSystem() string {
-	var operatingSystem string
+// operatingSystemResult caches the result of operatingSystem below, for efficiency
+// There's no need to protect this variable with a lock: the worst that can happen
+// here is several threads competing to set the value at the start - but they'll all
+// come up with the same result, and setting a pointer is an atomic operation
+var operatingSystemResult *string
+
+func operatingSystem() (operatingSystem string) {
+	if operatingSystemResult != nil {
```

Comment:
This does not look safe.
There can be multple calls here and end up with two goroutines trying to set the same value and corrupting memory.

Reply:
@cpuguy83 : that's addressed by the comment just above (lines 246-248). @thaJeztah seemed to agree there. Happy to discuss more if you think that comment is erroneous.

Reply:
Hmm... I don't think this comment is true.

Reply:
AFAIK the spec does not say that setting pointers is atomic even if it may be in practice, so we cannot rely on this.
But if we set the value in one place at daemon startup, this should be fine.

Reply:
Changed to using a mutex.

Reply:
Maybe the code will look easier if `sync.Once` was used?